### ISSUE TITLE
Can't load state with layout split in multiple RenderViews

### ISSUE THREAD

**Issue by fgava90** [Kitware/visualizer#90]
Hello, 

I am trying to use Visualizer to compare some data. 
I have created a .pvsm file in which i have my layout split in 3 RenderViews, One with the reference data, one with the data I want to compare and the other with the difference.
I have managed to use --load-file to load the state, and it seems like it's working, however on the browser i can see only the first RenderView.
If I use on-screen rendering I can see the Paraview window becoming larger, however the new part remains white.
Also, nothing shows up in the Pipeline Browser.
All this with Paraview-5.8.0 on Centos 8.

**Comment by jourdain** [comment]
Visualizer does not support multi-views. But if your goal is to just compare results using state files, it might be easier to just create a custom ParaViewWeb app that can deal with the multiple view.

Another issue that you might be running into is the path of the files that you load from the state file. If they are not correct, they will fail inside Visualizer.

HTH

**Comment by fgava90** [comment]
Thank you @jourdain , I was afraid of that. 
The data loaded by the state is inside the a subdirectory of the --data directory, so i supposed it would be able to find it.
I used a python script to produce the state and in it all paths to files are absolute.
However, I do use a reader and some filters that are not available in the UI of Visualizer.
Should I finally decide to develop a new app, or just to do testing with a single view on Visualizer, do I have to add such reader and filters to the app or, since the work is done by paraview in background, it's fine as it is? (As long as I do not need to use them from the browser of course)

**Comment by jourdain** [comment]
Are the absolute path still valid in the state file?
The state file can use any filter, even the one that are not shown to the user in the UI.
You should be able to use visualizer via a single view if that is enough for you.
And even if filters are not in the UI you will be able to control them.

**Comment by fgava90** [comment]
I have checked the state file and everything seems ok.
Then I have tried to arrive at my final state progressively and I still see stuff in the Pipeline until I use the ResampleWithDataset filter. 
If I load a state with this filter, the pipeline in Visualizer is empty.
My pipeline in Paraview is:
|- file1.case
|     |- CellDataToPointData1
|     |- MergeBlocks1
|     '- ResampleWithDataset1
|- file2.case
|     |- CellDataToPointData2
|     |- MergeBlocks2
|     '- ResampleWithDataset1
'-ResampleWithDataset1

**Comment by jourdain** [comment]
Are you using the same ParaView version to generate the state than the one running visualizer?
I'm guessing loading that state in ParaView just works?

**Comment by fgava90** [comment]
Yes, I have downloaded the precompiled Paraview-5.8 distribution and I am using the visualizer that comes with it.
And yes, loading the state in Paraview works fine.

**Comment by jourdain** [comment]
Do you have any output error? I'm not sure to know why  ResampleWithDataset is making the pipeline fail.

**Comment by fgava90** [comment]
The only output I have is:
`wslink: Starting factory
CRITICAL:twisted:wslink: Starting factory`
But I have it also when the Pipeline works. I am not sure whether there is a log file somewhere that i could check. 
Just in case, this is my command line:
/usr/local/opt/paraview-5.8.0/bin/pvpython --force-offscreen-rendering /usr/local/opt/paraview-5.8.0/share/paraview-5.8/web/visualizer/server/pvw-visualizer.py --content /usr/local/opt/paraview-5.8.0/share/paraview-5.8/web/visualizer/www/ --port 8080 --data /home/gavaf/data/ --load-file test_state.pvsm

**Comment by jourdain** [comment]
It seems that is "working" without error. I really have no clue why you are getting nothing in your pipeline once you have the ResampleWithDataset filter.

**Comment by fgava90** [comment]
I guess it's just my bad luck.
I have tried to add the ResampleWithDataSet filter to Visualizer with the --proxies file.
If I try to apply the filter to my data I get this error (repeated several times):

( 201.555s) [main thread     ]       vtkExecutive.cxx:753    ERR| vtkPVCompositeDataPipeline (0xc58af50): Algorithm vtkPResampleWithDataSet(0xac36110) returned failure for request: vtkInformation (0xcafd4f0)
  Debug: Off
  Modified Time: 8700874
  Reference Count: 1
  Registered Events: (none)
  Request: REQUEST_DATA_OBJECT
  FROM_OUTPUT_PORT: 0
  ALGORITHM_AFTER_FORWARD: 1
  FORWARD_DIRECTION: 0

**Comment by jourdain** [comment]
It seems that your data loading is failing and nothing gets loaded. Are you sure the state file load successfully without telling ParaView to use other paths than the one in the state file?

**Comment by fgava90** [comment]
I have tried applying a calculator to the data loaded with the state file and it worked.
I have also tried to manually load some new data directly in Visualizer. I can apply other filters but not ResampleWitDataset.
Could it be because ResampleWithDataset should open a popup window to manually select the mesh data?

**Comment by jourdain** [comment]
I didn't know the ResampleWithDataset was pop-ing a window. Then yes, that make sense except that with the state file everything should be all-set and therefore should not require any pop-up window.

**Comment by fgava90** [comment]
Exactly my thought.
Could it be that the problem is that this filter takes inputs from two different "branches" of the pipeline, so the representation gets broken?

**Comment by jourdain** [comment]
Yes that must be the problem.
